### PR TITLE
API-1465 diagnostic report V2

### PR DESCRIPTION
**Pull request by jblefkove-va**
https://vajira.max.gov/browse/API-1456

Depends on https://github.com/department-of-veterans-affairs/health-apis-data-query/pull/555

### CODE REVIEW

File: db/migration/V5_3__create_diagnostic_report_v2.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  [DateUTC] smalldatetime NULL,
+  [LastUpdated] smalldatetime default getutcdate(),
+  [DiagnosticReport] varchar(max) NOT NULL
+  constraint PK_DiagnosticReport_V2 primary key clustered (CDWId)
+)
+GO
+
+CREATE INDEX [IX_DiagnosticReport_V2_PatientFullICN] on [App].[DiagnosticReport_V2]([PatientFullICN])
+GO
+
```
Comment (jhulbert-va) [inline db/migration/V5_3__create_diagnostic_report_v2.sql:15]: Should there be indexes on category and code as well?

File: minimart-manager/pom.xml [hunk 2]
```diff
@@ -11,7 +11,7 @@
   <version>1.0.0-SNAPSHOT</version>
   <packaging>jar</packaging>
   <properties>
-    <data-query.version>3.0.1</data-query.version>
+    <data-query.version>3.0.2-SNAPSHOT</data-query.version>
```
Comment (jhulbert-va) [inline minimart-manager/pom.xml:14]: Remove snapshot after DQ merge.

File: minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/MitreMinimartMaker.java [hunk 3]
```diff
@@ -229,6 +297,31 @@ private void insertByDiagnosticReport(List<File> files) {
...
+        file -> {
+          try {
+            DatamartDiagnosticReports wrapper =
+                JacksonConfig.createMapper().readValue(file, DatamartDiagnosticReports.class);
+            for (DatamartDiagnosticReports.DiagnosticReport report : wrapper.reports()) {
+              save(
+                  DiagnosticReportEntity.builder()
+                      .cdwId(report.identifier())
+                      .icn(dm.fullIcn())
+                      .category("LAB")
```
Comment (jhulbert-va) [inline minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/MitreMinimartMaker.java:311]: Always lab?

File: minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/MitreMinimartMaker.java [hunk 4]
```diff
@@ -229,6 +297,31 @@ private void insertByDiagnosticReport(List<File> files) {
...
+                      .code(null)
+                      .dateUtc(null)
+                      .lastUpdated(null)
```
Comment (jhulbert-va) [inline minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/MitreMinimartMaker.java:314]: Should we be populating these?

File: minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/MitreMinimartMaker.java (hunk 3 above)
Comment (jblefkove-va) [reply minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/MitreMinimartMaker.java:311]: That's my guess based on how DSTU2 diagnostic reports are set up. If you search by category, it's hard-coded to check for `LAB` and return nothing for anything else.

**Pull request merged**